lib/datasource/npm/npmrc.ts

```diff
@@ -112,8 +112,7 @@ export function resolvePackage(packageName: string): PackageResolution {
     !authInfo &&
     npmrc &&
     npmrc._authToken &&
-    registryUrl.replace(regEx(/\/?$/), '/') ===
-      npmrc.registry?.replace(/\/?$/, '/') // TODO #12070
+    registryUrl.replace(/\/?$/, '/') === npmrc.registry?.replace(/\/?$/, '/') // TODO #12070
```

Comment:
I am thinking of handling this one and similar changes will be handles in #12071.
The issue #12071 is almost irrelevant now since the regex is being cached, still regex like this and some others can be moved up in scope so that in future we don't need to change multiple instances of same regex.

Reply:
We have a special function for this though, wouldn't it be more readable to use it?

Reply:
Yes, I meant the same thing. Including this instance there are some more places where we need to add ensureTrailingSlash so I wanted to implement this in another PR.